### PR TITLE
Check task network device before querying network stats

### PR DESCRIPTION
**Pull request by shubham2892**
### Summary
- `StartStatsCollection` was being called for every container in a task, it should be called only once for every task.
- Task network device name was being fetched only at the beginning of collection of stats, it might be possible that the devices has not been attached to the container yet, instead the check is done everytime before querying for the task.

### Testing
 How was this tested?

Tested manually that `startStatsCollection` is called only once for pause container for an AWSVPC task. Also, Tested manually that device name is not none by running an AWSVPC task. Before this change, I could see instances of device name being none.

As a result of these two changes, I am able to consistently see right `rx_bytes_per_sec` when using v4 taskmeta data endpoint, before this change,  `rx_bytes_per_sec` in lot of instances.

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/stats/engine.go [hunk 1]
```diff
@@ -179,8 +179,15 @@ func (engine *DockerStatsEngine) addAndStartStatsContainer(containerID string) {
 	if err != nil {
 		return
 	}
+
+	dockerContainer, errResolveContainer := engine.resolver.ResolveContainer(containerID)
+	if errResolveContainer != nil {
+		return
```
Comment (yunhee-l) [inline agent/stats/engine.go:185]: should something be logged here?

**Comment by ubhattacharjya** [comment]
Can you add the testing done explanation in the PR?
